src/raw/mod.rs

```diff
@@ -299,15 +308,15 @@ impl<T> Bucket<T> {
             // Just return an arbitrary ZST pointer which is properly aligned
             mem::align_of::<T>() as *mut T
         } else {
-            self.ptr.as_ptr()
+            self.ptr.as_ptr().sub(1)
```

Comment:
Could you add a comment on the `ptr` member of `Bucket` saying that this actually points to the next element and that this is needed to maintain pointer arithmetic invariants.

Reply:
Done.